streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -170,28 +167,31 @@ public boolean initializeStateStores() {
     @Override
     public void initializeTopology() {
         initTopology();
+
+        // if (eos) begin new transaction
```

Comment:
Nit: this comment should go into JavaDoc  or be removed -- it's clear from the code what is happening and the comment does not add anything IMHo.